### PR TITLE
/dart-2: Add better intros, change title

### PR DESCRIPTION
**Pull request by kwalrath**
I need to do a pass over referring pages to make sure link titles &
references are OK.

Staged at: https://kw-www-dartlang-1.firebaseapp.com/dart-2
/cc @anders-sandholm, @mit-mit, @kevmoo

### CODE REVIEW

**Comment by kevmoo** [comment]
Nice!

File: src/dart-2.md [hunk 1]
```diff
@@ -1,8 +1,47 @@
 ---
-title: Dart 2 Migration
+title: Dart 2
 description: How Dart 2 is different from Dart 1.x, and how you can convert your code to work with Dart 2.
 ---
 
+Dart 2 has a few key differences from earlier versions of Dart.
```
Comment (chalin) [inline src/dart-2.md]: By "version" we usually mean major version (IMHO), so "from earlier versions" seems strange to me.

> Dart 2 has a few key differences from Dart 1.x.

File: src/dart-2.md [hunk 2]
```diff
@@ -1,8 +1,47 @@
...
+This page briefly covers those differences and
+gives general advice on migrating your code to Dart 2.
+Also see the migration guide for your platform:
+
+* [Flutter migration instructions][]
```
Comment (mit-mit) [inline src/dart-2.md]: https://github.com/flutter/flutter/wiki/Dart-2-Migration

**Comment by kwalrath** [comment]
PTAL. I've updated the staged version:

https://kw-www-dartlang-1.firebaseapp.com/dart-2

Also check out the version drop-down (at the upper right).

**Pull request merged**